### DISCUSSION TITLE
Gas and Water meter Water Meter Reading Volume per half rotation

### DISCUSSION THREAD

**Discussion by jac2424** [tronikos/esphome-magnetometer-water-gas-meter#14] [General]
For accuracy of Volume per half rotation parameter, would it be better to set volume_unit setting to match the unit used by the meter rather than do conversions (my water and gas meters are both in m3)?

Also to determine the correct number to use would below be the appropriate calculation?

Water meter reading 1: 3662.125 m3
Water meter reading 2: 3662.69 m3

Water used: 0.565 m3

Water Meter Reading Half rotations total 1: 81413
Water Meter Reading Half rotations total 2: 91832

Water Meter Reading Half rotations: 10,419

Volume per half rotation calculation:

0.563 m3 / 10,419 = 0.00005403589 m3  per half rotation

So the number I would enter for  Volume per half rotation in home assistant config would be 0.00005403589 m3 without any further need for conversion of units?

**Comment by tronikos** [comment]
I don't expect the accuracy to improve if you set the units to match. I'd set the unit based on your preference. To improve accuracy observe the values over a long period.

Yes your calculations seem fine.
